File: 1544-Make-The-String-Great.py
class Solution:
    def makeGood(self, s: str) -> str:
        n = len(s)
        stk = []

        for i in range(n):
            c = s[i]
            if stk and c.lower() == stk[-1].lower() and c != stk[-1]:
                stk.pop()
            else:
                stk.append(c)

        return "".join(stk)

File: neetcode/1544-Make-The-String-Great.ts
function makeGood(s: string): string {
  const stk: string[] = []

  for (const c of s) {
    const top = stk.at(-1)

    if (top && c !== top && c.toLowerCase() === top.toLowerCase()) stk.pop()
    else stk.push(c)
  }

  return stk.join("")
};
